websites/D/Discords.com/presence.ts

```diff
@@ -0,0 +1,175 @@
+const presence = new Presence({
+  clientId: "843791837273391104"
+}), browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "discords_logo",
+    startTimestamp: browsingStamp
+  }; 
+
+    if (document.location.pathname === "/") {
+      presenceData.details = "Viewing home page";
+    } else if (document.location.pathname === "/servers/search") {
+      const page = document.querySelector("li.page-item.active")?.textContent,
+      search = document.querySelector("h1.text-center")?.textContent.replace("Discord Servers", " ");
+      presenceData.details = `🔍 Searching for: ${search || "Nothing"}`;
+      presenceData.state = `📖 Page ${page}`;
+      presenceData.buttons = [
+        {
+          label: "View Results",
+          url: document.location.href
+        }
+      ];
+    } else if (document.location.pathname.includes("/servers/tag/")) {
+      const page = document.querySelector("li.page-item.active")?.textContent,
+      tag = document.querySelector("h1.text-center")?.textContent.replace("Discord Servers", " ");
+      presenceData.details = `Viewing 📛 ${tag || "Nothing"} tag`;
+      presenceData.state = `📖 Page ${page}`;
+      presenceData.buttons = [
+        {
+          label: "View Tag",
+          url: document.location.href
+        }
+      ];
+    } else if (document.location.pathname.includes("/servers/tags/")) {
+      const page = document.querySelector("li.page-item.active")?.textContent,
+      tag = document.querySelector("h1.text-center")?.textContent.replace("Discord Servers", " ");
+      presenceData.details = `Viewing 🔑 ${tag || "Nothing"} keyword`;
+      presenceData.state = `📖 Page ${page}`;
+      presenceData.buttons = [
+        {
+          label: "View Keyword",
+          url: document.location.href
+        }
+      ];
+    } else if (document.location.pathname === "/servers/top-100") {
+      const page = document.querySelector("li.page-item.active")?.textContent;
+      presenceData.details = "Viewing 🏅 top-100 servers";
+      presenceData.state = `📖 Page ${page}`;
+      presenceData.buttons = [
+        {
+          label: "View Top-100",
+          url: document.location.href
+        }
+      ];
+    } else if (document.location.pathname.includes("/join")) {
+      const server_name_join = document.querySelector("h2.mt-4")?.textContent;
+      presenceData.details = `Joining 🚦 ${server_name_join || "Nothing"} `;
+      presenceData.buttons = [
+        {
+          label: `Join ${server_name_join || "Nothing"}`,
+          url: document.location.href
+        }
+      ];
```

Comment:
Join nothing, yeah....

Reply:
Yeah, the only reason I did that is due to the discords.com website takes some time to load the data. So while people wait instead of undefined it displays "nothing" is that ok?

Reply:
if(server_name_join) presenceData.buttons